workflow/controller/steps.go

```diff
@@ -112,7 +112,8 @@ func (woc *wfOperationCtx) executeSteps(ctx context.Context, nodeName string, tm
 				// This happens when there was `withItem/withParam` expansion.
 				// We add the aggregate outputs of our children to the scope as a JSON list
 				var childNodes []wfv1.NodeStatus
-				for _, node := range woc.wf.Status.Nodes {
+				for _, childID := range sgNode.Children {
+					node := woc.wf.Status.Nodes[childID]
 					if node.BoundaryID == stepsCtx.boundaryID && strings.HasPrefix(node.Name, childNodeName+"(") && node.Type != wfv1.NodeTypeSkipped {
```

Comment:
Similar with this check? Also, I wonder why the corresponding code in `dag.go` does not have `node.Type != wfv1.NodeTypeSkipped`?

Reply:
Sorry, i need more time to review this code.